### PR TITLE
Themes: Add Pills Styling to Search

### PR DESCRIPTION
**Pull request by ockham**
Ripped from @budzanowski's M5 spike, #8282.

image

### CODE REVIEW

**Comment by folletto** [comment]
Note: we'll style in a separate PR.

Classes, I'd change slightly the names to:

* themes-magic-search-card__search-token → themes-magic-search-card__token
* themes-magic-search-card__search-taxonomy → themes-magic-search-card__token-taxonomy
* themes-magic-search-card__search-separator → themes-magic-search-card__token-separator
* themes-magic-search-card__search-filter → themes-magic-search-card__token-filter

Reason: the "search" bit is already in the root, and the "token" is a bit more general. :)

**Comment by folletto** [comment]
Tested on Chrome, Safari, Firefox: 👍 
 
Well done. :)

File: client/my-sites/themes/themes-magic-search-card/index.jsx [hunk 1]
```diff
@@ -71,15 +71,20 @@ const ThemesMagicSearchCard = React.createClass( {
...
 			tokens.map( ( token, i ) => {
-				let classname = 'themes-magic-search-card__search-text';
-
 				if ( token.trim() === '' ) {
-					classname = 'themes-magic-search-card__search-white-space';
+					return <span className="themes-magic-search-card__search-white-space" key={ i }>{ token }</span>; // use shortid for key
 				} else if ( filterIsValid( token ) ) {
-					classname = 'themes-magic-search-card__search-token';
+					const separator = ':';
+					const [ taxonomy, filter ] = token.split( separator );
```
Comment (ockham) [inline client/my-sites/themes/themes-magic-search-card/index.jsx]: @seear Should we be using `../theme-filters` `getTaxonomy` and `getTerm` instead?

Comment (seear) [inline client/my-sites/themes/themes-magic-search-card/index.jsx]: Yes, mainly to keep all that logic in one place.

Comment (ockham) [inline client/my-sites/themes/themes-magic-search-card/index.jsx]: Fun times, `getTerm`and `getTaxonomy` don’t wfm here. I type `color:blue` and as soon as I hit the `e`, that is swallowed.

Comment (seear) [inline client/my-sites/themes/themes-magic-search-card/index.jsx]: Merge as is, then. I'll check it out later.

**Pull request merged**